stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/externimpl/connection/CreateConnection.java

```diff
@@ -0,0 +1,114 @@
+        try {
+            @SuppressWarnings(ArtemisConstants.UNCHECKED)
+            BMap<String, BValue> connection = (BMap<String, BValue>) context.getRefArgument(0);
+
+            String url = context.getStringArgument(0);
+
+            @SuppressWarnings(ArtemisConstants.UNCHECKED)
+            BMap<String, BValue> configObj = (BMap<String, BValue>) context.getRefArgument(1);
+            long connectionTTL = ((BInteger) configObj.get("connectionTTL")).intValue();
+            long callTimeout = ((BInteger) configObj.get("callTimeout")).intValue();
+            int consumerWindowSize = ArtemisUtils.getIntFromConfig(configObj, "consumerWindowSize", logger);
+            int consumerMaxRate = ArtemisUtils.getIntFromConfig(configObj, "consumerMaxRate", logger);
+            int producerWindowSize = ArtemisUtils.getIntFromConfig(configObj, "producerWindowSize", logger);
+            int producerMaxRate = ArtemisUtils.getIntFromConfig(configObj, "producerMaxRate", logger);
+            long retryInterval = ((BInteger) configObj.get("retryInterval")).intValue();
+            double retryIntervalMultiplier = ((BFloat) configObj.get("retryIntervalMultiplier")).floatValue();
+            long maxRetryInterval = ((BInteger) configObj.get("maxRetryInterval")).intValue();
+            int reconnectAttempts = ArtemisUtils.getIntFromConfig(configObj, "reconnectAttempts", logger);
+            int initialConnectAttempts = ArtemisUtils.getIntFromConfig(configObj, "initialConnectAttempts", logger);
+
+            ServerLocator connectionPool = ActiveMQClient.createServerLocator(url);
+
+            //Add config values to the serverLocator before creating the sessionFactory
+            connectionPool.setConnectionTTL(connectionTTL);
+            connectionPool.setCallTimeout(callTimeout);
+            connectionPool.setConsumerWindowSize(consumerWindowSize);
+            connectionPool.setConsumerMaxRate(consumerMaxRate);
+            connectionPool.setProducerWindowSize(producerWindowSize);
+            connectionPool.setProducerMaxRate(producerMaxRate);
+            connectionPool.setRetryInterval(retryInterval);
+            connectionPool.setRetryIntervalMultiplier(retryIntervalMultiplier);
+            connectionPool.setMaxRetryInterval(maxRetryInterval);
+            connectionPool.setReconnectAttempts(reconnectAttempts);
+            connectionPool.setInitialConnectAttempts(initialConnectAttempts);
+            connectionPool.setConfirmationWindowSize(1024);
+
+            ClientSessionFactory factory = connectionPool.createSessionFactory();
+
+            connection.addNativeData(ArtemisConstants.ARTEMIS_CONNECTION_POOL, connectionPool);
+            connection.addNativeData(ArtemisConstants.ARTEMIS_SESSION_FACTORY, factory);
+
+        } catch (Exception e) {
+            ArtemisUtils.throwBallerinaException("Error occurred while starting connection.", context, e);
```

Comment:
Shouldn't it better if we log error from the class which the error is occurred rather than calling util class to log?